src/natcap/invest/sdr/sdr.py

```diff
@@ -557,15 +577,31 @@ def execute(args):
         args['biophysical_table_path'],
         **MODEL_SPEC['args']['biophysical_table_path'])
 
-    # Test to see if c or p values are outside of 0..1
-    for key in ['usle_c', 'usle_p']:
-        for lulc_code, row in biophysical_df.iterrows():
-            if row[key] < 0 or row[key] > 1:
-                raise ValueError(
-                    f'A value in the biophysical table is not a number '
-                    f'within range 0..1. The offending value is in '
-                    f'column "{key}", lucode row "{lulc_code}", '
-                    f'and has value "{row[key]}"')
+    # Test to see if C values are outside of 0..1
+    def update_out_of_range(any_out_of_range, block):
+        if numpy.any(block[numpy.any([block < 0, block > 1])]):
+            any_out_of_range = True
+        return any_out_of_range
+
+    any_out_of_range = pygeoprocessing.raster_reduce(
+        function=update_out_of_range,
+        raster_path_band=(args['usle_c_path'], 1),
+        initializer=False)
+
+    if any_out_of_range:
```

Comment:
It's worth noting that historically we haven't done validation on rasters like this. Checking that every pixel value meets some condition has the potential to add significant compute time on large datasets. I think I'm in support of it, but I'd be curious to get more opinions.

Reply:
It definitely doesn't belong in `validate`, because that needs to return near-instant results to the workbench. Maybe we should have a "`slow_validate`" function for validation steps that may take a long time, and the user could opt out of.